docs/docs/concepts/memory.md

```diff
@@ -303,8 +285,24 @@ def update_instructions(state: State, store: BaseStore):
 
 ![](img/memory/update-instructions.png)
 
-#### Few-shot examples
+## Writing memories
+
+While [humans often form long-term memories during sleep](https://medicine.yale.edu/news-article/sleeps-crucial-role-in-preserving-memory/), AI agents need a different approach. When and how should agents create new memories? There are at least two primary methods for agents to write memories: "on the hot path" and "in the background".
+
+![](img/memory/hot_path_vs_background.png)
+
+### Writing memories in the hot path
+
+Creating memories during runtime offers both advantages and challenges. On the positive side, this approach allows for real-time updates, making new memories immediately available for use in subsequent interactions. It also enables transparency, as users can be notified when memories are created and stored.
+
+However, this method also presents challenges. It increases complexity, as the agent must decide what to commit to memory alongside its primary task. This can impact performance, potentially slowing down response times due to additional decision-making and degrading tool-calling performance and task completion rates. Additionally, the multitasking involved may result in fewer memories being saved, affecting recall in future conversations.
```

Comment:
"It increases complexity, as the agent must decide what to commit to memory alongside its primary task."

I don't buy this. The agent already needs to know how to work with memory on the read path AND regardless the logic has to exist somewhere, so it just doesn't sound like a significant increase in complexity that's worth mentioning?

Reply:
This should be explained better: it means that an agent would need a tool (e.g., to write and / or format) memories, which is strictly an increase in overhead / complexity.

Reply:
> The agent already needs to know how to work with memory on the read

Arguably read is just a hard-coded call for the store rather than a tool that the agent needs to manage. But, I get the point that the argument needs to be a bit more crisp.